packages/mariadb/src/MariaDbConnection.ts

```diff
@@ -1,17 +1,19 @@
 import { Connection } from 'mariadb';
-import { MySqlConnectionConfig } from 'knex';
-import { MySqlConnection } from '@mikro-orm/mysql-base';
+import { MySqlConnection, Knex } from '@mikro-orm/mysql-base';
+import { Utils } from '@mikro-orm/core';
 
-// @ts-ignore
-import Dialect from 'knex/lib/dialects/mysql/index.js';
+const Dialect = Utils.requireFrom(
+	'knex/lib/dialects/mysql/index.js',
+	require.resolve('@mikro-orm/knex', { paths: [require.resolve('@mikro-orm/mysql-base')] })
```

Comment:
while i was a long time tabs advocate, this project uses spaces for indentation so lets be consistent :] 

```suggestion
  'knex/lib/dialects/mysql/index.js',
  require.resolve('@mikro-orm/knex', { paths: [require.resolve('@mikro-orm/mysql-base')] })
```